base/compiler/tfuncs.jl

```diff
@@ -717,7 +718,7 @@ function getfield_nothrow(@nospecialize(s00), @nospecialize(name), @nospecialize
         s.abstract && return false
         # If all fields are always initialized, and bounds check is disabled, we can assume
         # we don't throw
-        if bounds_check_disabled && !isvatuple(s) && s.name !== NamedTuple.body.body.name && fieldcount(s) == s.ninitialized
+        if bounds_check_disabled && !isvatuple(s) && s.name !== NamedTuple.body.body.name && fieldcount(s) === s.ninitialized
```

Comment:
This makes it more brittle:
```
julia> typeof(Int).ninitialized |> typeof
Int32

julia> fieldcount(Int) |> typeof
Int64
```

Reply:
Huh, TIL.
i thought both of these were certain to be `Int`